pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -90,6 +90,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	err := r.Get(ctx, req.NamespacedName, &externalSecret)
 	if apierrors.IsNotFound(err) {
 		syncCallsTotal.With(syncCallsMetricLabels).Inc()
+		conditionSynced := NewExternalSecretCondition(esv1alpha1.ExternalSecretDeleted, v1.ConditionFalse, esv1alpha1.ConditionReasonSecretDeleted, "Secret was deleted")
+		SetExternalSecretCondition(&externalSecret, *conditionSynced)
```

Comment:
The externalSecret has no `.Name` or `.Namespace` set because it was deleted and wasn't fetched - the resulting metric will have empty name/namespace labels.

Either we set the object meta before calling `NewExternalSecretCondition()` or we work with the `externalSecretCondition *prometheus.GaugeVec` directly.

Reply:
Hey @moolen thanks for the feedback. We could just pass the details when we call the `SetExternalSecretCondition ` right? check my latest commit.